### PR TITLE
libmemcached on darwin

### PR DESCRIPTION
**Pull request by maxxk**
Build memcached and libmemcached on darwin. Memcached builds without changes (recent 1.4.24 does not due to clang warnings). Libmemcached requires a patch used in Homebrew.

### CODE REVIEW

File: pkgs/development/libraries/libmemcached/libmemcached-yosemite-ntohll.patch [hunk 1]
```diff
@@ -0,0 +1,236 @@
+Originally from Homebrew:
+https://github.com/Homebrew/homebrew/blob/bfd4a0a4626b61c2511fdf573bcbbc6bbe86340e/Library/Formula/libmemcached.rb#L42
```
Comment (copumpkin) [inline pkgs/development/libraries/libmemcached/libmemcached-yosemite-ntohll.patch]: Couldn't you just use `fetchpatch` here? As much as possible, it's nice to leave upstream things upstream.

Comment (maxxk) [inline pkgs/development/libraries/libmemcached/libmemcached-yosemite-ntohll.patch]: Thank you for the suggestion. I haven't thought that patch will process the ruby code preceding diff as a description :) I've also added links to upstream issues.

**Comment by offlinehacker** [comment]
Looks good to me

**Comment by copumpkin** [comment]
@maxxk thanks for changing that!

**Pull request merged**